app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -2279,6 +2280,35 @@ class PluginApi {
   registerHashtagType(type, typeClassInstance) {
     registerHashtagType(type, typeClassInstance);
   }
+
+  /**
+   * Adds a button to the bulk topic actions modal.
+   *
+   * ```
+   * api.addBulkActionButton({
+   *   label: "super_plugin.bulk.enhance",
+   *   icon: "magic",
+   *   class: "btn-default",
+   *   visible() {
+   *     return this.siteSettings.super_plugin_enabled && this.currentUser.staff;
+   *   },
+   *   async action() {
+   *     await enhance(this.model.topics);
+   *     doSomethingElse();
+   *   },
```

Comment:
This action gets bound to the controller and then executed. That essentially means that the entire modal controller (and soon, the modal component) becomes 'public API'.

Could we avoid that by removing the `.bind`, and instead pass any needed params/callbacks into the action? e.g.

```js
async action({topics, performAndRefresh}){
   ...
}
```

(and similarly for the `visible` function)